Unix socket support

kdubb opened the pull request:
Adds support for unix domain sockets via Netty.

* Added `unixsocket` URL property to add a domain socket to the list of hosts.
  * Unix socket addresses are _prepended_ to the attempted address list; an assumption was made that if you are using a unix socket with TCP fallback the unix socket is preferred.
* Requires Netty native libraries to be on the classpath
  * Can be done by customizing `pom.xml` (see netty docs) but is not done by default.
* Needs to be added to travis configuration

kdubb committed:
Allow “any” character in conneciton URL pattern

Also, cleans it up a bit by removing unnecessary escapes

kdubb committed:
Support unix domain sockets via Netty native

Yep.. we can cannot via a unix domain socket with netty. Requires either kqueue or epoll native netty jars to be provided.

kdubb commented:
**NOTE**

The `unixsocket` property can be a directory or a specific unix domain socket file.  If the property points to a directory it attempts to locate a matching `.s.PGSQL.XXXX` file automatically.  If it finds multiple files it chooses the first one it finds (and logs a warning).

jesperpedersen commented:
Looks good, apart from your path in `pgjdbc-ng.iml` and missing update to `README.md`

kdubb committed:
Rid ourselves of iml files

kdubb committed:
Updat README with unixsocket parameter.

kdubb committed:
Try upper & lower case of enum settings

kdubb committed:
Default to using native if the libraries are avail

If the native netty libraries are on the classpath for the current OS we default to using them, and falling back to NIO; if somebody put them there they probably want to use them.

This lessens the likelihood that somebody will _forget_ to enable and mix them with domain sockets which would create 2 separate thread pools.
